pkg/tracing/tracing.go

```diff
@@ -67,16 +67,27 @@ type Tracer struct {
 	ot.Tracer
 
 	enabled bool
+
+	// this indirection exists strictly to enable error injection
+	inject func(sm ot.SpanContext, format interface{}, carrier interface{}) error
```

Comment:
I want us to be at 100% test coverage in the near future. I'm happy to hear of alternative approaches to do that in this case.

I consider this  a pretty cheap and expedient way to get the job done without having to create a whole bunch of mock infrastructure.